backend/src/schema/resolvers/passwordReset.spec.js

```diff
@@ -16,158 +21,203 @@ const getAllPasswordResets = async () => {
+        it('resolves anyways', async () => {
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+            data: { requestPasswordReset: true },
+          })
+        })
+
+        it('creates no node', async () => {
+          await mutate({ mutation, variables })
+          const resets = await getAllPasswordResets()
+          expect(resets).toHaveLength(0)
+        })
       })
-    })
 
-    describe('with a valid email', () => {
-      const variables = { email: 'user@example.org' }
+      describe('with a valid email', () => {
+        beforeEach(() => {
+          variables = { ...variables, email: 'user@example.org' }
+        })
 
-      it('resolves', async () => {
-        await expect(client.request(mutation, variables)).resolves.toEqual({
-          requestPasswordReset: true,
+        it('resolves', async () => {
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+            data: { requestPasswordReset: true },
+          })
         })
-      })
 
-      it('creates node with label `PasswordReset`', async () => {
-        await client.request(mutation, variables)
-        const resets = await getAllPasswordResets()
-        expect(resets).toHaveLength(1)
-      })
+        it('creates node with label `PasswordReset`', async () => {
+          let resets = await getAllPasswordResets()
+          expect(resets).toHaveLength(0)
+          await mutate({ mutation, variables })
+          resets = await getAllPasswordResets()
+          expect(resets).toHaveLength(1)
+        })
 
-      it('creates a reset code', async () => {
-        await client.request(mutation, variables)
-        const resets = await getAllPasswordResets()
-        const [reset] = resets
-        const { code } = reset.properties
-        expect(code).toHaveLength(6)
+        it('creates a reset nonce', async () => {
+          await mutate({ mutation, variables })
+          const resets = await getAllPasswordResets()
+          const [reset] = resets
+          const { nonce } = reset.properties
+          expect(nonce).toHaveLength(6)
+        })
       })
     })
   })
+})
+
+describe('resetPassword', () => {
+  const setup = async (options = {}) => {
+    const { email = 'user@example.org', issuedAt = new Date(), nonce = 'abcdef' } = options
 
-  describe('resetPassword', () => {
-    const setup = async (options = {}) => {
-      const { email = 'user@example.org', issuedAt = new Date(), code = 'abcdef' } = options
+    const session = driver.session()
+    await createPasswordReset({ driver, email, issuedAt, nonce })
+    session.close()
+  }
 
-      const session = driver.session()
-      await createPasswordReset({ driver, email, issuedAt, code })
-      session.close()
+  const mutation = gql`
+    mutation($nonce: String!, $email: String!, $newPassword: String!) {
+      resetPassword(nonce: $nonce, email: $email, newPassword: $newPassword)
     }
+  `
+  const nonce = 'abcdef'
```

Comment:
why do you have a default and then set the same value to a variable?

Reply:
so this calls `createPasswordReset` with the nonce `abcdef` as a default, then uses that same nonce to find the `PasswordReset` node, but if the user passes in a different nonce like `slkdjf`, then it will create a `PasswordReset` node with that `nonce`, but will not match the `resetPassword` nonce?